Upgrade Gradle to 7.4 & AGP to 7.1.1

ParaskP7 opened the pull request:
This PR upgrades `Gradle to 7.4` & `AGP to 7.1.1`.

It includes the following:
- `Gradle` version upgraded to `7.3.3` with the `./gradlew wrapper --gradle-version=7.3.3 --distribution-type=all` command.
- `AGP` version upgrade to `7.0.4` (see `WordPress/build.gradle` change).
- Build output diff:
  - To identify and fix new warnings/errors that got added.
  - To identify and verify old warnings/errors that got removed.
- Lint output diff
  - To identify and fix new warnings/errors that got added.
  - To identify and verify old warnings/errors that got removed.
- The above 4 steps were redone with a follow-up update to `Gradle 7.4` & `AGP 7.1.1`.
- [Update buildkite docker android build image to 1.2.0.](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/d792317cda3cad7cb81c96d7e4500c1f38bd6f77)

-----

It suppresses the following Lint issues:
- [Missing default resources for aztec night colors.](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/14a86b4362cd9db4ac34a9a014978dadbb9376cd)
- [Missing default resources for editor dimens.](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/adbcbc7ab88314239c49951f00c959fdd6c55a63)
- [Notification trampoline for post upload notifier.](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/f58b114a446ed7dd668d164b6f74b7fc492202be)
- [Unused resources for dimens.](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/263e1c094509b75f742e5d6ca0b0ea46845ad73f)
- [Unused resources for strings.](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/addcfd200ab98f622c6b4eea563904048f883f82)
- [Unused resources for styles.](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/423342e6abd07ffaff29e1c9d216836927bc5556)
- [Unused resources for widget login flow toolbar style.](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/1b5d160f6c72a2031b83ae26b4028a414cbfb505)
- [Protected permissions for android manifest permission.](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/c9702a42c842a3ac80023954a7b4b6b50c6c45e9)

It resolves the following Lint issues:
- [Obsolete sdk int version checks for wordpress module.](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/692ab28a49cce7c0e540b30ab6ae96ed6b2d1879)
- [Unused resources for dimens.](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/263e1c094509b75f742e5d6ca0b0ea46845ad73f)
- [Outdated lint new api warning path xml configuration.](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/c318c4bb598c2b4bc7c4afee4885fc0aa2634124)
- [Restricted api for quick start prompt dialog fragment.](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/43449cebddb38aba824f7b7aedebe72bbd8bf348)

It created the following GitHub issues:
- [Android 12 - Launch Activity directly from the Notification #15875](https://github.com/wordpress-mobile/WordPress-Android/issues/15875)
- [Lint - Remove Tag Soup Library and Html To Spanned Converter #15882](https://github.com/wordpress-mobile/WordPress-Android/issues/15882) ✅
- [Gradle - Remove Obsolete 'dexOptions' Element #15885](https://github.com/wordpress-mobile/WordPress-Android/issues/15885)
- [Gradle - Determine Obsolete 'BaseVariant.getApplicationIdTextResource' Call #15886](https://github.com/wordpress-mobile/WordPress-Android/issues/15886)
- [Gradle - Unable to Strip Some Libraries, Packaging as They Are #15887](https://github.com/wordpress-mobile/WordPress-Android/issues/15887)
- [Gradle - Some Annotation Processors are Not Incremental #15888](https://github.com/wordpress-mobile/WordPress-Android/issues/15888)

-----

Besides the above, you can reference the `Gradle 7.4 & AGP 7.1.1` branches of the below libraries for testing/verification purposes:
- [x] [WordPress-FluxC-Android](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2258)
- [x] [gutenberg](https://github.com/WordPress/gutenberg/pull/39508) + [gutenberg-mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4662)
- [x] [stories-android](https://github.com/Automattic/stories-android/pull/719)
- [x] [AztecEditor-Android](https://github.com/wordpress-mobile/AztecEditor-Android/pull/963)
- [x] [about-automattic-android](https://github.com/Automattic/about-automattic-android/pull/34)
- [x] [WordPress-Login-Flow-Android](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/83)
- [x] [WordPress-Utils-Android](https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/105)

**NOTE**: This is a draft PR because all `Gradle & AGP` upgrade PRs need to be merged together.

## Merge Instructions

- [x] Update all 7 library references to PR hash (this is done for testing purposes):
  - [x] [fluxCVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/900cfe12b8bcbb54fe19618247b26a5ec00478e0)
  - [x] [gutenbergMobileVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/46cb24f02228ec7ed5d87ec17cd49ea4d441b64a)
  - [x] [storiesVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/4397510a4ac3cae0beb720557ce7d108455f58f1)
  - [x] [aztecVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/557a8e646a2c8c91a97fbb810d975f2840e0788e)
  - [x] [aboutAutomatticVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/0d12999c2245beb0bb7577397e3e4c34dea5a398)
  - [x] [wordPressLoginVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/630507dd38da3f91d78f2c1cc5df3d4fe99a7584)
  - [x] [wordPressUtilsVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/9be09db398aecb5ce93b445ab90156bbb81808f5)
- [x] Mark PR as `Ready for review` and assign @wordpress-mobile/owl-team to it.
- [x] Merge all 7 libraries into `trunk` (after them being code reviewed and approved)
  - [x] [WordPress-FluxC-Android](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2258)
  - [x] [gutenberg](https://github.com/WordPress/gutenberg/pull/39508) + [gutenberg-mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4662) (Kudos to @fluiddot for that! 🙇 )
  - [x] [stories-android](https://github.com/Automattic/stories-android/pull/719)
  - [x] [AztecEditor-Android](https://github.com/wordpress-mobile/AztecEditor-Android/pull/963)
  - [x] [about-automattic-android](https://github.com/Automattic/about-automattic-android/pull/34)
  - [x] [WordPress-Login-Flow-Android](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/83)
  - [x] [WordPress-Utils-Android](https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/105)
- [x] Update all 7 library references to `trunk` hash or tag:
  - [x] [fluxCVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/a93f23568eae9dcaf95adabdf7db9e2fba9f1c52)
  - [x] [gutenbergMobileVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/ff0e10b348d40744892b8ce9a4fb084a90d8a33e)
  - [x] [storiesVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/19dcd3ce8e16dc4e242c0528fc336a593c2f4a4c)
  - [x] [aztecVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/ad55fc804176a3d2a4f021b50245cedfcb42a42a)
  - [x] [aboutAutomatticVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/55ad1f20a72525e0c625be23e30137434f31daeb)
  - [x] [wordPressLoginVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/215046a73c5276d3fa51904846505eeca46695ad)
  - [x] [wordPressUtilsVersion](https://github.com/wordpress-mobile/WordPress-Android/pull/15865/commits/eb5d81963697bb3ccb67df407f601ec041ca4368)
- [x] Wait for PR to get approved.
- [x] Create a new `gutenbergMobileVersion` alpha tag and use it as the reference. (`FYI: Never merge trunk branch refeferences or commit SHAs, only tags.`)
- [x] Remove `[Status] Not Ready for Merge` label.
- [x] Merge this PR.

## To Test

- Smoke test the app.
- Use composite build for all 7 libraries (following the instructions in [local-builds.gradle-example](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/local-builds.gradle-example) file) and smoke test the app again. PS: Make sure you're on the `Gradle & AGP` branches on all 7 libraries.
- On the Gutenberg side of things, please note the below:
  - Run `nvm use` from the `gutenberg-mobile/gutenberg` folder (otherwise you might have errors).
  - Run `npm install` from the `gutenberg-mobile` folder (to download and install the project dependencies).
  - Run `npm run start:reset` from the `gutenberg-mobile` folder (to start the metro server).

## Regression Notes

1. Potential unintended areas of impact

N/A - The build file changes should only impact whether the project builds or not. However, since AGP is updated, a few Lint issues got fixed, it's hard to be sure that nothing else will be impacted.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual tests and CI checks.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ParaskP7 committed:
Build: Update gradle to 7.3.3

ParaskP7 committed:
Build: Update agp to 7.0.4

ParaskP7 committed:
Analysis: Suppress missing default resources for aztec night colors

ParaskP7 committed:
Analysis: Suppress missing default resources for editor dimens

ParaskP7 committed:
Analysis: Suppress notification trampoline for post upload notifier

This is currently suppressed because:
- Currently, this isn't causing any crashes or the app misbehaving.
This issue can just lead to significant lags in some scenarios.
- Also, it is not trivial to fix this and thus out of scope of the
Gradle and AGP upgrade. The activity should be directly launched from
the notification.

However, this issue will need to be fixed sooner than later, especially
because of the fact that it will become forbidden when the
'targetSdkVersion' will be set to API level for Android 12 or higher.
Thus, a GitHub issue was created to accompany this commit and issue
suppression. From more info see the newly created GitHub issue below:

https://github.com/wordpress-mobile/WordPress-Android/issues/15875

ParaskP7 committed:
Analysis: Resolve obsolete sdk int version checks for wordpress module

ParaskP7 committed:
Analysis: Resolve/Suppress unused resources for dimens

ParaskP7 committed:
Analysis: Suppress unused resources for strings

ParaskP7 committed:
Analysis: Suppress unused resources for styles

ParaskP7 committed:
Analysis: Suppress unused resources for widget login flow toolbar style

ParaskP7 committed:
Analysis: Remove outdated lint new api warning path xml configuration

ParaskP7 committed:
Analysis: Suppress lint error for html to spanned converter

The lint detector androidx.appcompat
.BaseMethodDeprecationDetector$ApiAbove called context.getMainProject()
during module analysis.

This does not work correctly when running in AGP (7.0.4).

In particular, there may be false positives or false negatives because
the lint check may be using the minSdkVersion or manifest information
from the library instead of any consuming app module.

Contact the vendor of the lint issue to get it fixed/updated (if
known, listed below), and in the meantime you can try to work around
this by disabling the following issues:

Call stack: Context
 .getMainProject(Context.kt:117)
←BaseMethodDeprecationDetector$ApiAbove
 .matches(BaseMethodDeprecationDetector.kt:58)
←BaseMethodDeprecationDetector$DeprecationCondition
 .matches(BaseMethodDeprecationDetector.kt:136)
←BaseMethodDeprecationDetector
 .visitMethodCall(BaseMethodDeprecationDetector.kt:158)
←UElementVisitor$DelegatingPsiVisitor
 .visitMethodCallExpression(UElementVisitor.kt:1128)
←UElementVisitor$DelegatingPsiVisitor
 .visitCallExpression(UElementVisitor.kt:1108)
←UCallExpression$DefaultImpls
 .accept(UCallExpression.kt:85)
←UCallExpressionEx$DefaultImpls
 .accept(UCallExpression.kt:-1)
←JavaUCallExpression
 .accept(javaUCallExpressions.kt:26)
←UQualifiedReferenceExpression$DefaultImpls
 .accept(UQualifiedReferenceExpression.kt:34)
←JavaUCompositeQualifiedExpression
 .accept(JavaUCompositeQualifiedExpression.kt:23)
←UVariableKt
 .visitContents(UVariable.kt:64)
←UVariableKt
 .access$visitContents(UVariable.kt:1)
←ULocalVariable$DefaultImpls
 .accept(UVariable.kt:110)
←ULocalVariableEx$DefaultImpls
 .accept(UVariable.kt:-1)
←JavaULocalVariable
 .accept(JavaUVariable.kt:92)
←ImplementationUtilsKt
 .acceptList(implementationUtils.kt:23)
←UDeclarationsExpression$DefaultImpls
 .accept(UDeclarationsExpression.kt:22)
←JavaUDeclarationsExpression
 .accept(JavaUDeclarationsExpression.kt:7)
←ImplementationUtilsKt
 .acceptList(implementationUtils.kt:23)

This issue type represents a problem running lint itself. Examples
include failure to find bytecode for source files (which means certain
detectors could not be run), parsing errors in lint configuration
files, etc.

These errors are not errors in your own code, but they are shown to make
it clear that some checks were not completed.

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Tmp: Remove tag soup library and html to spanned converter

This commit will help identify why Lint is failing on CI due to the
[HtmlToSpannedConverter] java class.

PS: This is a temporary, to-be-reverted commit.

ParaskP7 committed:
Revert "Tmp: Remove tag soup library and html to spanned converter"

This reverts commit 8552141e

ParaskP7 committed:
Analysis: Resolve get color state list for html to spanned converter

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Build: Update gradle to 7.4

ParaskP7 committed:
Build: Update agp to 7.1.1

ParaskP7 committed:
Analysis: Suppress protected permissions for android manifest permission

ParaskP7 committed:
Analysis: Resolve restricted api for quick start prompt dialog fragment

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
CI: Update buildkite docker android build image to 1.2.0

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

 Conflicts:
	WordPress/src/main/res/values/strings.xml

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

 Conflicts:
	WordPress/src/main/java/org/wordpress/android/util/HtmlToSpannedConverter.java

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

ParaskP7 committed:
Build: Update gutenberg mobile hash

This 'Gutenberg Mobile' hash updates 'Gutenberg Mobile' to that branch
version where 'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'Gutenberg' changes
work as expected with this WPAndroid client.

Gutenberg PR: https://github.com/WordPress/gutenberg/pull/39508

ParaskP7 committed:
Build: Update fluxc hash

This 'FluxC' hash updates 'FluxC' to that branch version where
'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'FluxC' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Build: Update login hash

This 'Login' hash updates 'Login' to that branch version where
'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'Login' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Build: Update stories hash

This 'Stories' hash updates 'Stories' to that branch version where
'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'Stories' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Build: Update aztec hash

This 'Aztec' hash updates 'Aztec' to that branch version where
'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'Aztec' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Build: Update about hash

This 'About' hash updates 'About' to that branch version where
'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'About' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Build: Update utils hash

This 'Utils' hash updates 'Utils' to that branch version where
'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'Utils' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

 Conflicts:
	build.gradle

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

 Conflicts:
	build.gradle

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

 Conflicts:
	WordPress/src/main/res/values/strings.xml
	build.gradle

ParaskP7 committed:
Fix: Add missing build import to gcm message handler

oguzkocer commented:
> AGP version upgrade to 7.0.4 (see android/settings.gradle.kts change).

@ParaskP7 I think this is a copy paste mistake as we don't have a `settings.gradle.kts` file in the project and the AGP version is set in the root `build.gradle` file.

ParaskP7 commented:
👋 @oguzkocer !

> @ParaskP7 I think this is a copy paste mistake as we don't have a settings.gradle.kts file in the project and the AGP version is set in the root build.gradle file.

Nice catch, you are right, thank you, I updated the description to `(see 'WordPress/build.gradle' change)`! 🙇

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

oguzkocer committed:
Always use react-native-mirror repository to fetch react-native

ParaskP7 committed:
Build: Update fluxc hash

This 'FluxC' hash updates 'FluxC' to that branch version where
'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'FluxC' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

 Conflicts:
	build.gradle

ParaskP7 committed:
Build: Update gutenberg mobile tag

This 'Gutenberg Mobile' alpha tag updates 'Gutenberg Mobile' to that
version where 'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'Gutenberg' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Build: Update fluxc hash

This 'FluxC' hash updates 'FluxC' to that trunk version where
'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'FluxC' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Build: Update login hash

This 'Login' hash updates 'Login' to that trunk version where
'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'Login' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Build: Update stories hash

This 'Stories' hash updates 'Stories' to that trunk version where
'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'Stories' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Build: Update aztec hash

This 'Aztec' hash updates 'Aztec' to that trunk version where
'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'Aztec' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Build: Update about hash

This 'About' hash updates 'About' to that trunk version where
'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'About' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Build: Update utils hash

This 'Utils' hash updates 'Utils' to that trunk version where
'Gradle & AGP' is also upgraded to version '7.4 & 7.1.1'
correspondingly.

This step is required in order to check that these 'Utils' changes
work as expected with this WPAndroid client.

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into update/gradle-to-7.3.3-agp-to-7.0.4

 Conflicts:
	build.gradle
